File: algorithms/M/Minimum Subsequence in Non-Increasing Order/Minimum Subsequence in Non-Increasing Order.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var minSubsequence = function(nums) {
  const target = nums.reduce((a, b) => a + b) / 2;
  nums.sort((a, b) => b - a);
  let i = 0, sum = 0;
  while (sum <= target) {
    sum += nums[i++];
  }
  return nums.slice(0, i);
};

File: scripts/algorithms/M/Minimum Subsequence in NonIncreasing Order/Minimum Subsequence in Non-Increasing Order.java
// Runtime: 7 ms (Top 34.08%) | Memory: 44.9 MB (Top 74.04%)
class Solution {
    public List<Integer> minSubsequence(int[] nums) {
        int total = 0;
        for(int i=0;i<nums.length;i++){
            total += nums[i];
        }
        Arrays.sort(nums);
        int sum = 0;
        ArrayList<Integer> ans = new ArrayList<>();
        for(int i=nums.length-1;i>=0;i--){
            ans.add(nums[i]);
            sum += nums[i];
            if(sum>total-sum){
                return ans;
            }
        }
        return ans;
    }
}